test.py

```diff
@@ -0,0 +1,180 @@
+# pyGene Fusion local testing
+import json
+import re
+from pydantic import BaseModel, validator, ValidationError, StrictStr, StrictInt, StrictBool
+from typing import Optional, List, Union
+
+class GeneDescriptor(BaseModel):
+    """Define a Gene, denoted by GeneDescriptor"""
+    type = 'GeneDescriptor'
+    value_id: str
+    label: StrictStr
+
+    @validator('value_id')
+    def is_valid(cls,v):
+        assert v.count(':') == 1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class SequenceLocation(BaseModel):
+    """Define SequenceLocation class"""
+    type = 'SequenceLocation'
+    seq_id: str
+    start: StrictInt
+    end: StrictInt
+
+    @validator('seq_id')
+    def is_curie(cls, v):
+        assert v.count(':') == 1 and v.find(' ') == -1, 'chr must be a CURIE'
+        return v
+
+class ChromosomeLocation(BaseModel):
+    """Define ChromosomeLocation class"""
+    type = 'ChromosomeLocation'
+    species: str
+    chr: str
+    start: str
+    end: str
+
+    @validator('species')
+    def is_curie(cls, v):
+        assert v.count(':') == 1 and v.find(' ') == -1, 'chr must be a CURIE'
+        return v
+
+    @validator('chr')
+    def valid_chr(cls, v):
+        assert v.isalnum(), 'chr must have only alphanumeric characters'
+        return v
+
+    @validator('start', 'end')
+    def valid_loc(cls, v):
+        assert bool(re.match(r"^cen|[pq](ter|([1-9][0-9]*(\.[1-9][0-9]*)?))$", v)) == True, \
+            r'start/end positions must match the regular expression ^cen|[pq](ter|([1-9][0-9]*(\.[1-9][0-9]*)?))$'
+        return v
+
+class GenomicRegion(BaseModel):
+    """Define GenomicRegion class"""
+    type = 'computed'
```

Comment:
Can you explain again why `type = "computed"` rather than `type = "GenomicRegion"`?

Reply:
Ok. So rather than setting it to computed, I would suggest just changing to `type: str` and you can add more validations once we know the types we'll have?